.github/workflows/oidc-e2e-test.yml

```diff
@@ -0,0 +1,136 @@
+          yq -i ".config.dynamic.authc.openid_auth_domain.authentication_backend.type = \"noop\"" config.yml
+          echo "Installing demo configuration"
+          cd ../../plugins/opensearch-security/tools
+          chmod +x install_demo_configuration.sh
+          ./install_demo_configuration.sh -y
+          echo "Starting OpenSearch"
+          ../../../bin/opensearch &
+          timeout 300 bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' https://localhost:9200 -ku admin:admin)" != "503" ]]; do sleep 5; done'
+          echo "Setting up certificates"
+          chmod +x securityadmin.sh
+          ./securityadmin.sh -cd ../../../config/opensearch-security/ -icl -nhnv \
+          -cacert ../../../config/root-ca.pem \
+          -cert ../../../config/kirk.pem \
+          -key ../../../config/kirk-key.pem
+          timeout 300 bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' https://localhost:9200 -ku admin:admin)" != "200" ]]; do sleep 5; done'
+      - name: Get and run OpenSearch-Dashboards
+        run: |
+          echo "Downloading OpenSearch-Dashboards ${{ env.VERSION }}"
+          wget https://artifacts.opensearch.org/releases/bundle/opensearch-dashboards/${{ env.VERSION }}/opensearch-dashboards-${{ env.VERSION }}-linux-x64.tar.gz
+          echo "Unpacking OpenSearch-Dashboards"
+          tar -xzf opensearch-dashboards-${{ env.VERSION }}-linux-x64.tar.gz
+          echo "Injecting OIDC configuration"
+          cd opensearch-dashboards-${{ env.VERSION }}/config
+          echo "opensearch_security.openid.connect_url: \"http://localhost:8080/auth/realms/master/.well-known/openid-configuration\"" >> opensearch_dashboards.yml
+          echo "opensearch_security.openid.client_id: \"opensearch\"" >> opensearch_dashboards.yml
+          echo "opensearch_security.openid.client_secret: \"${{ env.KEYCLOAK_CLIENT_SECRET }}\"">> opensearch_dashboards.yml
+          echo "opensearch_security.auth.type: [\"openid\"]" >> opensearch_dashboards.yml
+          echo "opensearch_security.auth.multiple_auth_enabled: true" >> opensearch_dashboards.yml
+          echo "opensearch_security.ui.openid.login.buttonname: \"OIDC\"" >> opensearch_dashboards.yml
+          echo "home.disableWelcomeScreen: true" >> opensearch_dashboards.yml
+          echo "Starting OpenSearch-Dashboards"
+          cd ../bin
+          chmod +x opensearch-dashboards
+          ./opensearch-dashboards serve &
+      - name: Checkout functional-test
+        uses: actions/checkout@v2
+        with:
+          repository: ${{github.repository}}
+          path: functional-test
+      - name: Get Cypress version
+        id: cypress_version
+        run: |
+          echo "::set-output name=cypress_version::$(cat ./functional-test/package.json | jq '.devDependencies.cypress' | tr -d '"')"
+      - name: Cache Cypress
+        id: cache-cypress
```

Comment:
Just for my knowledge, is there any reason we need for cache-cypress here?

Reply:
I just followed other workflows, we can remove it then 👍

Reply:
Oh I mean, if the other workflows have it it should be fine~